docs/javascripts/exp_calculator.js

```diff
@@ -0,0 +1,79 @@
+function postStatsGained(exp, stat) {
+  const POST_MAX = 500000;
+  const C = POST_MAX * 0.5;
+  return Math.floor(positiveQuadraticRoot(1, -1, -(exp / C + stat ** 2 - stat)));
+}
+
+function preStatsGained(hp, multiplier, exp, stat) {
+  const C = hp * multiplier * 0.5;
+  return Math.floor(positiveQuadraticRoot(1, -1, -(exp / C + stat ** 2 - stat)));
+}
+
+function positiveQuadraticRoot(a, b, c) {
+  return (-b + Math.sqrt(b ** 2 - 4 * a * c)) / (2 * a);
+}
+
+/**
+ * Calculate the stats gained based off of HP and transcendent status
+ * @param {Number} hp
+ * @param {Number} multiplier
+ * @param {Number} exp
+ * @param {Number} stat
+ * @param {Number} max
+ * @returns {Number}
+ */
+function statsGained(hp, multiplier, exp, stat, max) {
```

Comment:
This is just the math workings I came up with when working it out into code:
```
positiveQuadraticRoot = -b/2a + sqrt(b^2 - 4ac)/2a
negativeQuadraticRoot = -b/2a - sqrt(b^2 - 4ac)/2a

M = FINAL
N = CURRENT
MULTIPLIER = 17 OR 20
HP = BaseHP
EXP = Experience Boxed
POST_MAX = 500000

C = (HP * MULTIPLIER OR POST_MAX) * 0.5;
EXP = C(M - N)(M + N - 1)
EXP = C(M(M + N - 1) - N(M + N - 1))
EXP = C(M^2 + M*N - M - N^2 - M*N + N)
EXP/C = M^2 + M*N - M - M*N - N^2 + N
M^2 - M - (EXP/C + N^2 - N) = 0
```

M is the final result of the experience applied when the stat is N

Reply:
Looks good!